### PR TITLE
[Merged by Bors] - chore: generalise `MeasurableSpace (AddChar A M)`

### PR DESCRIPTION
**Pull request by YaelDillies**
This was ungeneralised in #18494, but I need this for `A` a finite group and `M = ℂ`, which is certainly not discrete.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by b-mehta** [comment]
@loefflerd Does this change give the wrong measurable space for structures you care about? Given that we're now insisting `Finite A`, the case you mentioned of `AddChar ℤ ℝ` is still not given a measurable space instance here.

**Comment by loefflerd** [comment]
Yes, this is fair enough I guess, since if `A` is a finite monoid then any `AddChar A M` will factor through a finite subset of `M` anyway, so there is no other sensible choice of measurable-space / topology / etc structure but the discrete one.

File: Mathlib/MeasureTheory/Constructions/AddChar.lean [hunk 1]
```diff
@@ -9,8 +9,7 @@ import Mathlib.MeasureTheory.MeasurableSpace.Defs
 /-!
 # Measurable space instance for additive characters
 
-This file endows `AddChar A M` with the discrete measurable space structure whenever `A` and `M`
-are themselves discrete measurable spaces.
+This file endows `AddChar A M` with the discrete measurable space structure whenever `A` is a finite discrete measurable space.
```
Comment (b-mehta) [inline Mathlib/MeasureTheory/Constructions/AddChar.lean:12]:
```suggestion
This file endows `AddChar A M` with the discrete measurable space structure whenever `A` is a finite
discrete measurable space.
```

**Review by b-mehta — COMMENTED** [review]
bors d+

**Pull request closed without merging**